### ISSUE TITLE
Russian translation

### ISSUE THREAD

**Issue by mgeisler** [google/comprehensive-rust#326]

**Comment by mgeisler** [comment]
For reference, there is a Russian translation of the Rust Book: https://github.com/rust-lang-ru/book.

**Comment by baltuky** [comment]
@mgeisler Would appreciate if you could assign this issue to me

**Comment by mgeisler** [comment]
> @mgeisler Would appreciate if you could assign this issue to me

There you go! :smile:

**Comment by mgeisler** [comment]
Hi @baltuky, would you be able to add a nearly-empty PO file for Russian? Just translate a few entries — that will get us started and it will make it easier for others to start contributing!

**Comment by baltuky** [comment]
Hey Martin,
Yes, I've created a PO file for Russian. I will create a PR for that later today.

**Comment by baltuky** [comment]
@mgeisler I've created a [PR](#677) that brings a fresh `ru.po` file, PTAL

**Comment by atmsk** [comment]
I would like to contribute. How do I get started?

**Comment by baltuky** [comment]
> I would like to contribute. How do I get started?

Hey 👋 
It makes sense to start by reading the [Translations of Comprehensive Rust 🦀](https://github.com/google/comprehensive-rust/blob/main/TRANSLATIONS.md) page. 

I've prepared a draft of `Day 1: Morning` part and working on `Day 1: Afternoon`. So, feel free to pick any other chapters.

**Comment by istolga** [comment]
also want to contribute

**Comment by mgeisler** [comment]
Hey all! There seems to be great momentum here! I propose that you **[split the work via a spreadsheet](https://docs.google.com/spreadsheets/d/1T9l_Gd4O7qe3rA5EnK4Rk-7pRO4TutukCuJdakUHDss/edit?usp=sharing).** That seems to work well for the two Chinese translations.

**Comment by baltuky** [comment]
> Hey all! There seems to be great momentum here! I propose that you **[split the work via a spreadsheet](https://docs.google.com/spreadsheets/d/1T9l_Gd4O7qe3rA5EnK4Rk-7pRO4TutukCuJdakUHDss/edit?usp=sharing).** That seems to work well for the two Chinese translations.
> 
> Edit it all you want, use as much or as little as you like!

That is a great idea. 💯  I've updated the spreadsheet with my assignments. Now the translation process should be more transparent.

**Comment by starostinak** [comment]
Hey, I also want to contribute 👋 Will look though the translations guide and pick something unclaimed from the spreadsheet.

**Comment by starostinak** [comment]
I've claimed day 2 morning in the spreadsheet.

**Comment by gorodnev** [comment]
hello, I'm interested in this activity as well. I'd take the 'bare-metal' chapter of the book.

**Comment by mgeisler** [comment]
Thanks @starostinak and @gorodnev! I suggest starting small to het familiar with the infrastructure.

**Comment by Makarevich** [comment]
I just checked [the assignment sheet](https://docs.google.com/spreadsheets/d/1T9l_Gd4O7qe3rA5EnK4Rk-7pRO4TutukCuJdakUHDss/edit#gid=1124000687) and realized that the list of files in the column A is quite obsolete and I suspect that some of the translations need to be updated as well. (There are certainly quite a few "fuzzy" entries in the ru.po, which need to be updated.) Would it make sense to track the assignments by chapter instead: Intro, Day 1: Morning, Day 1: Afternoon, etc?

**Comment by mgeisler** [comment]
Hey folks, just a heads up that we're tinkering with publishing the course as a PDF — including for the translations: #1572.

If you have input on the font to use, please let us know!

**Comment by jnovikov** [comment]
Hi. I want to contribute to this effort. Should I just add myself to the sheet and assign some pages there ?

**Comment by henrif75** [comment]
> Hi. I want to contribute to this effort. Should I just add myself to the sheet and assign some pages there ?

Hi @jnovikov!
You can follow the instructions on https://github.com/google/comprehensive-rust/blob/main/TRANSLATIONS.md.
I believe the assignment sheet is obsolete, given it was unchanged for quite a while and the Comprehensive Rust material was substantially updated. My suggestion to you is to either update the sheet or you start a PR, adding here what are you intending to do.
Thank you!

**Comment by 1dimir** [comment]
Hi! It seems that I'm in. And I have a few questions. 

What is the desired granularity of pull requests? From my point of view the translation task could be split into three stages. First one is to make all pages readable in Russian: translate everything that should be translated. Second, re-read, refine and rephrase section-wise. Third, collect some meta-information, remove obsolete translations and flags. So is it OK to pull really small changes until we start refining? Or I just try and see how it goes?

`PO` files format supports [translator comments](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html). I started using them quite extensively. Noting, why was a particular translation used, what possible alternatives were there etc. It may seem to duplicate pull request discussions, but having it in `po` comments will keep the context for future reviews. E.g. I'd also put there some summary of [concurrency terminology discussion from past changes](https://github.com/google/comprehensive-rust/pull/724#pullrequestreview-1457119972). Am I not misusing the tool feature? 

What is the right place to ask tiny questions regarding the source material? If it isn't worth starting a new Discussion. E.g. we have two *Hello, World!* titles, one is about the first program and another could have a title: *Hello Rust* as it's about the language introduction.

**Comment by mgeisler** [comment]
> `PO` files format supports [translator comments](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html). I started using them quite extensively. Noting, why was a particular translation used, what possible alternatives were there etc. It may seem to duplicate pull request discussions, but having it in `po` comments will keep the context for future reviews. E.g. I would also put there some summary of [concurrency terminology discussion from past changes](https://github.com/google/comprehensive-rust/pull/724#pullrequestreview-1457119972). Am I not misusing the tool feature?

I think that's a great idea! In general, I've tried to not be prescriptive in how people use the PO files, but I think it's a nice idea to document past discussions like you suggest.

Longer-term, I think we should switch to a translation platform, #1305. That might remove the need for PO files completely if it becomes the source of truth. I could also imagine letting the PO files be the source of truth and then synchronize them to the platform (and make PRs to the PO files when exporting from the platform). Regardless, most platforms will likely support some form of translation comments.

**Comment by mgeisler** [comment]
> So is it OK to pull really small changes until we start refining?

Yes, I would encourage that! I've seen in other languages that it's much easier to review and merge small PRs.

There are a few challenges in translating the course: a big one is the human aspect of finding someone who is motivated to do the work of translating and reviewing. Making the PRs small helps here since it makes it easier to merge uncontroversial translations.

> What is the right place to ask tiny questions regarding the source material? If it isn't worth starting a new Discussion. E.g. we have two _Hello, World!_ titles, one is about the first program and another could have a title: _Hello Rust_ as it's about the language introduction.

You are welcome to create issues for things that look like bugs or where you have ideas for improvements — discussions are probably more for open-ended, well, discussions. We don't have a super strict separation :smile: 

You're also very welcome to simply create a PR whenever you spot something weird in the source text. I'm not a native English speaker, so there are still some typos and inconsistencies hidden here and there.
